src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeCustomAttributeData.cs

```diff
@@ -1573,14 +1573,41 @@ private static void AddCustomAttributes(
                                 }
                             }
 
-                            RuntimePropertyInfo? property = (RuntimePropertyInfo?)(type is null ?
-                                attributeType.GetProperty(name) :
-                                attributeType.GetProperty(name, type, Type.EmptyTypes)) ??
+                            RuntimeMethodInfo? setMethod = null;
+                            RuntimeMethodInfo? getMethod = null;
+                            RuntimePropertyInfo? property = null;
+                            RuntimePropertyInfo? firstNotNullPropInHierarchy = null;
```

Comment:
This is a suspicious name. It is marked nullable, but called "not null". I would just call this `firstPropInHierarchy`.